### PR TITLE
Resolve special coverage

### PR DESCRIPTION
**Pull request by mparent61**
Adds new centralized "tunic_campaign_id", will eventually rename to campaign_id once migrated. 

TODO
- [x]  add filters

### CODE REVIEW

**Comment by benghaziboy** [comment]
LGTM

File: bulbs/api/views.py [hunk 1]
```diff
@@ -466,11 +468,63 @@ def list(self, request):
                 raise Http404("No content found matching UUID")
 
             content = get_object_or_404(Content, pk=match.kwargs.get('pk'))
-            return Response(ContentSerializer().to_representation(content))
+            return Response(ContentSerializer(content).data)
```
Comment (mparent61) [inline bulbs/api/views.py]: Just misc cleanup

Comment (benghaziboy) [inline bulbs/api/views.py]: no one was calling you on this. Don't be so defensive

File: bulbs/api/views.py [hunk 2]
```diff
@@ -466,11 +468,63 @@ def list(self, request):
...
+class SpecialCoverageResolveViewSet(viewsets.ReadOnlyModelViewSet):
+    """Retrieve special coverage containing specified content"""
+    model = SpecialCoverage
+
+    def list(self, request):
+        content_id = get_query_params(self.request).get("content_id")
+        if content_id:
+
+            content = get_object_or_404(Content, pk=content_id)
+            # TODO: Filters:
```
Comment (MichaelButkovic) [inline bulbs/api/views.py]: now done?

File: bulbs/api/views.py (hunk 1 above)
Comment (mparent61) [inline bulbs/api/views.py]: who invited you to this PR anyway

Comment (benghaziboy) [inline bulbs/api/views.py]: well that really hurts

**Comment by mparent61** [comment]
OK @benghaziboy @MichaelButkovic   ready for review

File: bulbs/api/views.py [hunk 3]
```diff
@@ -466,11 +468,61 @@ def list(self, request):
...
+                doc_type=search_objects.mapping.doc_type,
+                id=content.id,
+                body=special_coverage_filter,
+            )
+            # Translate perocolator results into SpecialCoverage objects
+            if results["total"]:
+                special_coverage_ids = [int(m["_id"].split(".")[-1]) for m in results["matches"]]
+                qs = SpecialCoverage.objects.filter(id__in=special_coverage_ids)
+
+                # Active Filter
```
Comment (mparent61) [inline bulbs/api/views.py]: Best way to do these filters? I'm not filtering via percolator, thinking it's better to just do it in ORM query.

File: bulbs/api/views.py [hunk 4]
```diff
@@ -466,11 +468,61 @@ def list(self, request):
...
+                active = get_query_params(self.request).get('active', '').lower()
+                now = timezone.now()
+                if active == 'true':
+                    qs = qs.filter(start_date__lte=now, end_date__gte=now)
+                elif active == 'false':
+                    qs = qs.exclude(start_date__lte=now, end_date__gte=now)
+
+                # Sponsored Filter
+                sponsored = get_query_params(self.request).get('sponsored', '').lower()
```
Comment (benghaziboy) [inline bulbs/api/views.py]: we can probably move these to the model or something so we aren't repeating the logic everywhere. But I'm working on some SC refactor stuff atm, so I can take a stab at it.

Comment (mparent61) [inline bulbs/api/views.py]: Good idea. The SpecialCoverageFilter could be refactored to share this logic too.

File: bulbs/api/views.py (hunk 3 above)
Comment (benghaziboy) [inline bulbs/api/views.py]: I almost feel like these could be bumped up to a manager method or something

File: tests/api/test_resolve_api.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                                   start_date=(timezone.now() + timedelta(days=1)),
+                                   end_date=(timezone.now() + timedelta(days=2)),
+                                   )
+        self.make_special_coverage(91, 'obama',
+                                   start_date=(timezone.now() - timedelta(days=1)),
+                                   end_date=(timezone.now() + timedelta(days=1)),
+                                   )
+        self.make_content(id=123, tags=['joe-biden', 'obama'])
+        r = self.resolve(content_id=123, active=False)
+        six.assertCountEqual(self, [90], [r['id'] for r in r.data])
```
Comment (mparent61) [inline tests/api/test_resolve_api.py]: Required the six version bump

**Comment by MichaelButkovic** [comment]
LGTM

File: bulbs/api/views.py (hunk 3 above)
Comment (mparent61) [inline bulbs/api/views.py]: Agreed. We'll refactor SC filters as a followup PR

**Pull request merged**